File: rust/tcl-registry/src/commands/irules/classify__disable.rs
//! `CLASSIFY::disable` iRules command.
use crate::prelude::*;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "CLASSIFY::disable",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Disables the classification of the flow.",
            synopsis: &["CLASSIFY::disable"],
            snippet: "Disables the classification of the flow",
            source: "https://clouddocs.f5.com/api/irules/CLASSIFY__disable.html",
            examples: "",
            return_value: "",
        }),
        forms: &[FormSpec {
            kind: FormKind::Default,
            synopsis: "CLASSIFY::disable",
        }],
        side_effects: &[SideEffect {
            target: SideEffectTarget::ClassificationState,
            reads: false,
            writes: true,
            connection_side: ConnectionSide::Both,
        }],
        ..CommandSpec::DEFAULT
    }
}

File: dialects/f5/irules/classify__disable.py
# Enriched from F5 iRules reference documentation.
"""CLASSIFY::disable -- Disables the classification of the flow."""

from __future__ import annotations

from compiler.registry._base import CommandDef
from compiler.registry.models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from compiler.registry.namespace_models import EventRequires
from compiler.registry.signatures import Arity
from compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget

from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/CLASSIFY__disable.html"


@register
class ClassifyDisableCommand(CommandDef):
    name = "CLASSIFY::disable"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="CLASSIFY::disable",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Disables the classification of the flow.",
                synopsis=("CLASSIFY::disable",),
                snippet="Disables the classification of the flow",
                source=_SOURCE,
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="CLASSIFY::disable",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.CLASSIFICATION_STATE,
                    writes=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )
